src/server/channel_store.cc

```diff
@@ -23,6 +23,30 @@ bool Matches(string_view pattern, string_view channel) {
   return stringmatchlen(pattern.data(), pattern.size(), channel.data(), channel.size(), 0) == 1;
 }
 
+// Build functor for sending messages to connection
```

Comment:
so the underlying buffer of messages must live until this function runs?
i am a bit confused why do you need to copy AND to pass the original messages array.